custom/motioncontrollers/CustomPIDController.java

```diff
@@ -350,4 +352,64 @@ public double get() {
 			return 0;
 		}
 	}
+
+	/**
+	 * Put the PID controller to SmartDashboard for tuning.
+	 * Puts the error, setpoint, sensor value, and output.
+	 * This method adds a very small random value to everything
+	 * so that graphs show properly on SmartDashboard.
+	 * 
+	 * @param prefix
+	 *        The prefix to use when putting things on SmartDashboard.
+	 */
+	public void putToSmartDashboard(String prefix) {
+		double noise = (Math.random() - 0.5) * 0.0000001; // Generate very small noise centered at zero
+		SmartDashboard.putNumber(prefix + "_Error", getError() + noise);
+		SmartDashboard.putNumber(prefix + "_Setpoint", getSetpoint() + noise);
+		SmartDashboard.putNumber(prefix + "_Sensor", getSensorValue() + noise);
+		SmartDashboard.putNumber(prefix + "_Output", get() + noise);
+	}
+
+	/**
+	 * Put the PID controller to SmartDashboard for tuning.
+	 * Puts the error, setpoint, sensor value, and output.
+	 * This method adds a very small random value to everything
+	 * so that graphs show properly on SmartDashboard.
+	 */
+	public void putToSmartDashboard() {
+		putToSmartDashboard(CustomPIDController.DEFAULT_SMARTDASHBOARD_PREFIX);
+	}
+
+	/**
+	 * Update the PID controller from SmartDashboard for tuning.
+	 * Gets the P, I, D, and F constants from SmartDashboard,
+	 * or puts the current ones there if there's no constants.
+	 * 
+	 * @param prefix
+	 *        The prefix to use when putting things on SmartDashboard.
+	 */
+	public void updateFromSmartDashboard(String prefix) {
+		// If we're missing any constants on SmartDashboard, put the current ones
+		if (!(SmartDashboard.containsKey(prefix + "_P") && SmartDashboard.containsKey(prefix + "_I")
+			&& SmartDashboard.containsKey(prefix + "_D") && SmartDashboard.containsKey(prefix + "_F"))) {
+			SmartDashboard.putNumber(prefix + "_P", getP());
+			SmartDashboard.putNumber(prefix + "_I", getI());
+			SmartDashboard.putNumber(prefix + "_D", getD());
+			SmartDashboard.putNumber(prefix + "_F", getD());
+			return;
+		}
+		setPIDF(SmartDashboard.getNumber(prefix + "_P", getP()),
```

Comment:
I think this is a good way of doing it, it will default to the previous P/I/D/F value.